columbo/_cli.py

```diff
@@ -45,6 +45,7 @@ def parse_args(
     :raises CliException: A value passed to CLI argument was not valid and `exit_on_error` was `False`.
     :raises DuplicateQuestionNameException: One of the given questions attempts to reuse a name. This includes a
         question that was used to create `answers` if given.
+    :raises ValueError: A default value was not valid and `no_user_input` is set to `True`.
```

Comment:
Question out of curiosity:

What is your approach to determine whether or not a function's docstring should say that it raises an exception when the function itself doesn't raise the exception, but one of the functions it calls may raise the exception? For example, in this instance, the `parse_args` function won't raise a ValueError directly, but the [`validate_duplicate_question_names` function it calls](https://github.com/wayfair-incubator/columbo/blob/3d89377998ac004571c1c9b84f9b25275ee1bdee/columbo/_cli.py#L50) may do so.

I see different approaches in different libraries and am curious for your thoughts.

Reply:
My thought is that it is better to list exceptions than not. A function that explicitly raises a function, that should be documented. If a function is calling a function that documents that it raises an exception, that should be documented (assuming the calling function can't do anything to prevent the called function from reaching the error state). Not particularly relevant here, but things start to head into a gray area for exceptions that are more unpredictable like `IOEror`.

This review made it clear that I haven't been doing the best at keeping up with that. The specific line here needs to be changed because I ended up not making this `no_user_input` specific. I'll check all the ones directly related to this PR now. However, I'll do a full pass in a follow-up PR.